### PR TITLE
fix(@rjsf/core): correctly omit extra data when additionalProperty is an object (#2667)

### LINKED ISSUE #2667: Passing additionalProperties does not work with omitExtraData
https://github.com/rjsf-team/react-jsonschema-form/blob/63636b7af2189bc151dfaa261318f167d6bb36e6/packages/core/src/utils.js#L1041

Passing additionalProperties = false does not work because in the line above. It should check if it is true instead of checking the precence:

http://json-schema.org/understanding-json-schema/reference/object.html#additional-properties

### PR DESCRIPTION
**Pull request by lpillonel**
### Reasons for making this change

fixes #2667 

When additionalProperties are an object, the data omission triggered with `liveOmit` and `omitExtraData` was skipped due to the flag set here :
https://github.com/rjsf-team/react-jsonschema-form/blob/36ee173e35f4859a624b7620426e9a009dafe811/packages/core/src/utils.js#L1142

Solution is to ensure that additionalProperties value is `true` in order to set the flag

### Checklist

* [x] **I'm updating documentation**
  - [x] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [ ] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/core/src/utils.js [hunk 1]
```diff
@@ -1139,7 +1139,10 @@ export function toPathSchema(schema, name = "", rootSchema, formData = {}) {
     return toPathSchema(_schema, name, rootSchema, formData);
   }
 
-  if (schema.hasOwnProperty("additionalProperties")) {
+  if (
+    schema.hasOwnProperty("additionalProperties") &&
+    schema.additionalProperties === true
+  ) {
```
Comment (epicfaace) [inline packages/core/src/utils.js:1145]: Thanks can you add a test?

Comment (lpillonel) [reply packages/core/src/utils.js:1145]: Test added; I amended the previous commit

**Comment by lpillonel** [comment]
@epicfaace 
Do you still expect anything to get this merged ?

**Comment by heath-freenome** [comment]
@lpillonel There is another [fix](https://github.com/rjsf-team/react-jsonschema-form/pull/2631/files) that is checking if `additionalProperties` is not false. It sounds like the more correct fix is yours?

**Comment by heath-freenome** [comment]
Fixed in version 5 beta via https://github.com/rjsf-team/react-jsonschema-form/pull/2940

**Pull request closed without merging**